PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -124,152 +123,40 @@ struct HfTrackIndexSkimCreatorTagSelCollisions {
-        SETBIT(statusCollision, EventRejection::Trigger);
-        if (fillHistograms) {
-          registry.fill(HIST("hEvents"), 3 + EventRejection::Trigger);
-        }
-      }
-
-      if (useTimeFrameBorderCut && !collision.selection_bit(o2::aod::evsel::kNoTimeFrameBorder)) {
-        SETBIT(statusCollision, EventRejection::TimeFrameBorderCut);
-        if (fillHistograms) {
-          registry.fill(HIST("hEvents"), 3 + EventRejection::TimeFrameBorderCut);
-        }
-      }
-    }
-
     float centrality = -1.;
-    if constexpr (centEstimator != CentralityEstimator::None) {
-      if constexpr (centEstimator == CentralityEstimator::FT0A) {
-        centrality = collision.centFT0A();
-      } else if constexpr (centEstimator == CentralityEstimator::FT0C) {
-        centrality = collision.centFT0C();
-      } else if constexpr (centEstimator == CentralityEstimator::FT0M) {
-        centrality = collision.centFT0M();
-      } else if constexpr (centEstimator == CentralityEstimator::FV0A) {
-        centrality = collision.centFV0A();
-      } else {
-        LOGP(fatal, "Centrality estimator not set!");
-      }
-      if (centrality < centralityMin || centrality > centralityMax) {
-        SETBIT(statusCollision, EventRejection::Centrality);
-        if (fillHistograms) {
-          registry.fill(HIST("hEvents"), 3 + EventRejection::Centrality);
-        }
-      }
-    }
-
-    // vertex selection
-    selectVertex(collision, statusCollision);
+    const auto statusCollision = getHfCollisionRejectionMask<applyTrigSel, centEstimator>(collision, centrality, centralityMin, centralityMax, useSel8Trigger, triggerClass, useTimeFrameBorderCut, zVertexMin, zVertexMax, nContribMin, chi2Max);
 
-    // selected events
     if (fillHistograms) {
-      if (statusCollision == 0) {
-        registry.fill(HIST("hEvents"), 2);
-        registry.fill(HIST("hPrimVtxX"), collision.posX());
-        registry.fill(HIST("hPrimVtxY"), collision.posY());
-        registry.fill(HIST("hPrimVtxZ"), collision.posZ());
-        if constexpr (centEstimator != CentralityEstimator::None) {
+      monitorCollision(collision, statusCollision, hEvents, hPrimVtxZBeforeSel, hPrimVtxZAfterSel, hPrimVtxXAfterSel, hPrimVtxYAfterSel, hNContributorsAfterSel);
+      // additional centrality histos
+      if constexpr (centEstimator != o2::aod::hf_collision_centrality::None) {
+        if (statusCollision == 0) {
           registry.fill(HIST("hCentralitySelected"), centrality);
-        }
-      } else {
-        if constexpr (centEstimator != CentralityEstimator::None) {
-          if (TESTBIT(statusCollision, EventRejection::Centrality)) {
-            registry.fill(HIST("hCentralityRejected"), centrality);
-          }
+        } else if (statusCollision == BIT(EventRejection::Centrality)) { // rejected by centrality only
+          registry.fill(HIST("hCentralityRejected"), centrality);
```

Comment:
The aim here is to check the centrality distribution of good events (either selected by the centrality cut or rejected). For instance we are not interested to check the centrality distribution of events that are not selected by `sel8` (also because the centrality calibration is done for events that have `sel8`).

Reply:
Done!